crates/matrix-sdk-ui/src/timeline/tests/read_receipts.rs

```diff
@@ -826,3 +826,39 @@ async fn test_threaded_latest_user_read_receipt() {
     assert_eq!(receipt_event_id, event_id!("$3"));
     assert_eq!(receipt.thread, receipt_thread);
 }
+
+#[async_test]
+async fn test_unthreaded_client_updates_threaded_read_receipts() {
+    let timeline = TestTimelineBuilder::new()
+        .settings(TimelineSettings { track_read_receipts: true, ..Default::default() })
+        .focus(TimelineFocus::Live { hide_threaded_events: true })
+        .build();
+    let mut stream = timeline.subscribe().await;
+
+    let event_b = event_id!("$event_b");
+
+    // Alice sends 2 messages
+    let f = &timeline.factory;
+    timeline.handle_live_event(f.text_msg("A").sender(*ALICE)).await;
+    timeline.handle_live_event(f.text_msg("B").sender(*ALICE).event_id(event_b)).await;
```

Comment:
I'm curious to propose a small change in this test, that I think wouldn't require a change in the rest of the code, in theory :)

Let's add a third event with id `event_c`, that's sent by Alice *in a thread*.

This event should be filtered out, because the timeline is set to `hide_threaded_events`.

Now, make Bob's read receipt point to `event_c`.

I think the outcome should remain the same, and the read receipt on `event_b` should be set to Bob's.

(This enhanced version proves that, even if the unthreaded read receipt is on an in-thread event, we can put it "at the right position" in the no-threaded-events timeline.)